src/JustSaying/Messaging/Channels/SubscriptionGroups/SubscriptionConfigBuilder.cs

```diff
@@ -38,6 +39,11 @@ public SubscriptionConfigBuilder()
         /// </summary>
         public TimeSpan DefaultReceiveBufferReadTimeout { get; private set; }
 
+        /// <summary>
+        /// Gets the default duration SQS will wait for a message before returning if there are no messages.
+        /// </summary>
```

Comment:
It's not just if there are no messages, it's if the `DefaultPrefetch` number of messages hasn't been fulfilled if that makes sense.
So with the defaults (20secs, 10 prefetch) if we get 9 messages, we'll still wait the full 20 seconds, but the moment we get 10 we stop and poll again for a new batch.

Reply:
Oh ok, I was going from the [docs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_ReceiveMessage.html#API_ReceiveMessage_RequestParameters) which say make it sound like it will return if one message is available

`WaitTimeSeconds = If a message is available, the call returns sooner than WaitTimeSeconds. If no messages are available and the wait time expires, the call returns successfully with an empty list of messages.`

`MaxNumberOfMessages  = The maximum number of messages to return. Amazon SQS never returns more messages than this value (however, fewer messages might be returned).`

Reply:
No, you are right @maisiesadler! I am recalling it wrong

Reply:
Ok :) will resolve the comment